frameaxis/semaxis.py

```diff
@@ -343,4 +340,37 @@ def word_contribution_to_second_moment(self, document, axis, corpus_mean, filter
             contrib = tf.multiply(tf.pow(diff, 2), frequencies_vec)/tf.reduce_sum(frequencies_vec)
             # kurtosis = moment4 / tf.square(moment2) - 3    
             contrib = sess.run([contrib])
-        return contrib, words     
\ No newline at end of file
+        return contrib, words     
+
+    def _prepare_contribution_mat_and_tf(self, documents, filter_stopword, min_freq):
+            vectorizer = CountVectorizer()
+            transformed_data = vectorizer.fit_transform(documents).toarray() #return dense matrix
+            frequencies = np.ravel(transformed_data.sum(axis=0))
+            
+            less_freq_idx = np.where(frequencies < min_freq)[0]
+            no_emb_idx = [idx for w, idx in vectorizer.vocabulary_.items() if w not in self.embedding.wv]
+            stopwords_idx = [idx for w, idx in vectorizer.vocabulary_.items() if filter_stopword and (w in stopwords.words('english'))]
+            to_filter_out = list(less_freq_idx) + no_emb_idx + stopwords_idx
+            freq_filtered = np.delete(transformed_data, to_filter_out, axis=1) # delete terms appearing less than min_freq
+            
+            self.doc_idx_to_filter = np.where(np.sum(freq_filtered,axis=1)<1)[0] # delete docs where all terms appear less than min_freq
+            terms_filtered = [self.embedding.wv[w] for w, idx in vectorizer.vocabulary_.items() if idx not in to_filter_out]
+            if self.axes_tfm_nn is None:
+                self.axes_mat = np.array([self.axes[k] for k in sorted(self.axes)])
+
+            return terms_filtered, freq_filtered
+
+    def _precompute_word_contribution_mat(self, documents, filter_stopword = True, min_freq = 10):
+        # get filtered terms and frequencies for all docs. Terms are filtered out if they
+        # appear less than min_freq times ACROSS ALL DOCS
+        self.terms_filtered, self.doc_freqs =  self._bao_prepare_matrix_computation(documents, filter_stopword, min_freq)  
```

Comment:
where is this function `_bao_prepare_matrix_computation()`'s body?

Reply:
oops I changed the name and forgot to refactor the rest. Change is pushed.